### DISCUSSION TITLE
adduser get error  request to http://localhost:4873/-/v1/login failed

### DISCUSSION THREAD

**Discussion by samkallon** [verdaccio/verdaccio#3672] [Q&A]
hello everyone
i am building private npm registry and i can visit  http://localhost:4873 
but when i use adduser i get the following error

npm ERR! FetchError: request to http://localhost:4873/-/v1/login failed, reason: connect ECONNREFUSED ::1:4873

here is the log , could anybody help? tks

0 verbose cli D:\nvm\nodejs\node.exe D:\nvm\nodejs\node_modules\npm\bin\npm-cli.js
1 info using npm@8.19.2
2 info using node@v18.12.1
3 timing npm:load:whichnode Completed in 1ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:D:\nvm\v18.12.1\node_modules\npm\npmrc Completed in 2ms
6 timing config:load:builtin Completed in 3ms
7 timing config:load:cli Completed in 3ms
8 timing config:load:env Completed in 1ms
9 timing config:load:file:D:\hthtProject\cesium_onemap\src\common\verdaccioLibTest\.npmrc Completed in 0ms
10 timing config:load:project Completed in 6ms
11 timing config:load:file:C:\Users\sam\.npmrc Completed in 1ms
12 timing config:load:user Completed in 1ms
13 timing config:load:file:D:\nodejs\node_global\etc\npmrc Completed in 0ms
14 timing config:load:global Completed in 1ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:credentials Completed in 1ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 19ms
19 timing npm:load:configload Completed in 19ms
20 timing npm:load:mkdirpcache Completed in 1ms
21 timing npm:load:mkdirplogs Completed in 0ms
22 verbose title npm adduser
23 verbose argv "adduser" "--auth-type" "legacy" "--registry" "http://localhost:4873"
24 timing npm:load:setTitle Completed in 1ms
25 timing config:load:flatten Completed in 3ms
26 timing npm:load:display Completed in 4ms
27 verbose logfile logs-max:10 dir:C:\Users\sam\AppData\Local\npm-cache\_logs
28 verbose logfile C:\Users\sam\AppData\Local\npm-cache\_logs\2023-03-10T07_23_29_686Z-debug-0.log
29 timing npm:load:logFile Completed in 6ms
30 timing npm:load:timers Completed in 0ms
31 timing npm:load:configScope Completed in 0ms
32 timing npm:load Completed in 32ms
33 warn adduser `adduser` will be split into `login` and `register` in a future version. `adduser` will become an alias of `register`. `login` (currently an alias) will become its own command.
34 notice Log in on http://localhost:4873/
35 verbose web login before first POST
36 silly logfile start cleaning logs, removing 7 files
37 timing command:adduser Completed in 26ms
38 verbose type system
39 verbose stack FetchError: request to http://localhost:4873/-/v1/login failed, reason: connect ECONNREFUSED ::1:4873
39 verbose stack     at ClientRequest.<anonymous> (D:\nvm\v18.12.1\node_modules\npm\node_modules\minipass-fetch\lib\index.js:130:14)
39 verbose stack     at ClientRequest.emit (node:events:513:28)
39 verbose stack     at Socket.socketErrorListener (node:_http_client:494:9)
39 verbose stack     at Socket.emit (node:events:525:35)
39 verbose stack     at emitErrorNT (node:internal/streams/destroy:151:8)
39 verbose stack     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
39 verbose stack     at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
40 verbose cwd D:\hthtProject\cesium_onemap\src\common\verdaccioLibTest
41 verbose Windows_NT 10.0.19044
42 verbose node v18.12.1
43 verbose npm  v8.19.2
44 error code ECONNREFUSED
45 error syscall connect
46 error errno ECONNREFUSED
47 error FetchError: request to http://localhost:4873/-/v1/login failed, reason: connect ECONNREFUSED ::1:4873
47 error     at ClientRequest.<anonymous> (D:\nvm\v18.12.1\node_modules\npm\node_modules\minipass-fetch\lib\index.js:130:14)
47 error     at ClientRequest.emit (node:events:513:28)
47 error     at Socket.socketErrorListener (node:_http_client:494:9)
47 error     at Socket.emit (node:events:525:35)
47 error     at emitErrorNT (node:internal/streams/destroy:151:8)
47 error     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
47 error     at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
47 error   code: 'ECONNREFUSED',
47 error   errno: 'ECONNREFUSED',
47 error   syscall: 'connect',
47 error   address: '::1',
47 error   port: 4873,
47 error   type: 'system'
47 error }
48 error
48 error If you are behind a proxy, please make sure that the
48 error 'proxy' config is set properly.  See: 'npm help config'
49 verbose exit 1
50 timing npm Completed in 96ms
51 verbose code 1
52 error A complete log of this run can be found in:
52 error     C:\Users\sam\AppData\Local\npm-cache\_logs\2023-03-10T07_23_29_686Z-debug-0.log

**Comment by juanpicado** [comment]
Advice, If you format the output (reading from mobile) and add all info you can get help otherwise ... good luck. Please check guidelines how to report issues or questions.

**Reply by jb747** [reply]
on my Mac, the default 'listen' setting in ~/.config/verdaccio/config.yaml is '0.0.0.0:4873'. On my system, using 'http://0.0.0.0:4873/' as the registry setting resolved the ECONNREFUSED error.
